internal/router/rpcservice/rpcservice.go

```diff
@@ -168,7 +173,31 @@ func (rs *RPCService) RunReadQuery(
 	if err != nil {
 		return RunReadQueryResponse{}, fmt.Errorf("calling RunReadQuery: %v", err)
 	}
-	return RunReadQueryResponse{Result: res}, nil
+	var opts []formatter.FormatOption
+	if req.Output != nil {
+		output, ok := formatter.OutputFromString(*req.Output)
+		if !ok {
+			return RunReadQueryResponse{}, fmt.Errorf("%s is not a valid output", *req.Output)
+		}
+		opts = append(opts, formatter.WithOutput(output))
+	}
+	if req.Extract != nil {
+		opts = append(opts, formatter.WithExtract(*req.Extract))
+	}
+	if req.Unwrap != nil {
+		opts = append(opts, formatter.WithUnwrap(*req.Unwrap))
+	}
+
+	formatted, config, err := formatter.Format(res, opts...)
+	if err != nil {
+		return RunReadQueryResponse{}, fmt.Errorf("formatting result: %v", err)
+	}
+
+	if config.Unwrap && len(res.Rows) > 1 {
+		return RunReadQueryResponse{}, errors.New("unwrapped results with more than one row aren't supported in JSON RPC API")
+	}
```

Comment:
Important decision here. In the case of would-be `jsonl` data ("unwrapped" data) with more than one row, I chose not to support it in the JSON RPC API. This is because it would be invalid JSON and the JSON RPC response must be valid JSON. If there is just one row, the unwrapped data is just a single JSON object and is valid JSON. The other alternative would be to base64 encode the unwrapped data and then many rows could be included. This just seemed like the more simple thing to start with. We can always do the base64 encoding later, or it sounds like we'll probably just delete the RPC service anyway.

Reply:
Makes sense. Would make sense to put this inside the `Format` and maybe eliminating the need to return the `config` value?

Reply:
The above would imply an extra error to make different returns according to the error.

I just saw the `user.go` file and it has the same logic.

Reply:
Hmm. The issue is that this logic equates to an error only in the context of the RPC service. In `user.go`, it's not considered an error, only that the content type header is set differently. Or maybe I'm not quite following what you're suggesting?

Reply:
Yea, see `user.go` it is:
```
		 if config.Unwrap && len(res.Rows) > 1 {
			rw.Header().Set("Content-Type", "application/jsonl+json")
		}
```

Reply:
Oh, I see. It's an error for RPC but not for HTTP. All good then.